libs/langchain-google-cloud-sql-pg/src/chatMessageHistory.ts

```diff
@@ -0,0 +1,143 @@
+import { BaseChatMessageHistory } from "@langchain/core/chat_history";
+import { AIMessage, BaseMessage, HumanMessage, StoredMessage, mapStoredMessagesToChatMessages } from "@langchain/core/messages";
+import PostgresEngine from "./engine.js";
+
+export interface PostgresChatMessageHistoryInput {
+  engine: PostgresEngine;
+  sessionId: string;
+  tableName: string;
+  schemaName: string;
+}
+
+export class PostgresChatMessageHistory extends BaseChatMessageHistory {
+  lc_namespace: string[] = ["langchain", "stores", "message", "google-cloud-sql-pg"];
+
+  engine: PostgresEngine;
+
+  sessionId: string;
+
+  tableName: string;
+
+  schemaName: string;
+
+  constructor({ engine, sessionId, tableName, schemaName = "public" }: PostgresChatMessageHistoryInput) {
+    super();
+    this.engine = engine;
+    this.sessionId = sessionId;
+    this.tableName = tableName;
+    this.schemaName = schemaName;
+  }
+
+  /**
+   * Create a new PostgresChatMessageHistory instance.
+   *
+   * @param {PostgresEngine} engine Postgres engine instance to use.
+   * @param {string} sessionId Retrieve the table content witht this session ID.
+   * @param {string} tableName Table name that stores that chat message history.
+   * @param {string} schemaName Schema name for the chat message history table. Default: "public".
+   * @returns PostgresChatMessageHistory instance.
+   */
+  static async create(
+    engine: PostgresEngine,
+    sessionId: string,
+    tableName: string,
+    schemaName: string = "public"
+  ) {
+    const query = `SELECT column_name, data_type FROM information_schema.columns WHERE table_name = '${tableName}' AND table_schema = '${schemaName}'`;
```

Comment:
May be worth a small note in docs emphasizing that `tableName` and `schemaName` in these files are not escaped and to not pass in end-user input here